docs/architecture/ADR-004-QGB_relayer_security.md

```diff
@@ -0,0 +1,82 @@
+
+## Alternative Approaches
+
+###  More  synchrony : Deploy the QGB contract with a data commitment window
+When deploying the QGB  contract,  also  set the data commitment window,  ie, the number of blocks between the `beginBlock` and `endBlock` of each data  commitment confirm.
+
+Then, update the QGB contract to check when receiving a new valset if the latest relayed data commitment height is >= new valset height - data commitment window.
+
+This also would mean adding, for example, a `DataCommitmentWindowConfirm` representing signatures of the validator set for a certain `DataCommitmentWindow`, since this latter can be updated using gov proposals.
+
+- Cons:
+	- More complexity and state in the contract
+- Pros:
+	- Fix the race condition issue
+
+### Add more state to the contract : Store valsets and their nonce
+Update the QGB contract to store the valset hashes + their nonces:
+- Cons:  
+		- Would make the contract more complex  
+- Pros:  
+		- Would make the relayer parallelizable (can submit data commitments and valsets in any order as long as the valset is committed)  
+		- would allow the QGB to catchup correctly even in the existence of a malicious relayer  
+
+### A request oriented design
+Currently, the attestations that need to be signed are defined by the state machine based on `abci.endBlock()` and a `DataCommitmentWindow`. This simplifies the state machine and doesn't require  implementing new transaction types to ask for orchestrators signatures.
+
+The request oriented design means providing users (relayers mainly) with the ability to post data commitment requests and ask orchestrators to sign them.
+
+The main issue with this approach is spamming and state bloat. In fact, allowing attestations signatures' requests would allow anyone to spam the network with unnecessary signatures and make orchestrators do unnecessary work. This gets worse if signatures are part of the state, since this latter is costly.
+
+A proposition to remediate the issues described above is to make the signatures part of the block data in specific namespaces. Then, we can charge per request and even make asking for attestations signatures a bit costly.
+
+- Pros
+	- Would allow anyone to ask for signatures over commitments, ie, the QGB can then be used by any team without changing anything.
+
+- Cons
+	- Makes slashing more complicated. In fact, to slash for liveness, providing the whole history of blocks, proving that a certain orchestrator didn't sign an attestation in the given period, will be hard to implement and the proofs will be big. Compared to the attestations being part of the state, which can be queried easilly.
+
+## Decision
+
+Still not made.
```

Comment:
barring something popping up, I think we can update this section now too

Reply:
check now please?